sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1984,3 +1985,33 @@ def test_warning_on_penalty_string_none():
     )
     with pytest.warns(FutureWarning, match=warning_message):
         lr.fit(iris.data, target)
+
+
+def test_lr_cv_scorer_does_not_receive_sample_weight_when_score_request_is_not_set():
```

Comment:
I quite like these two tests, but the tests you had in the common tests were also in the right direction. I don't think the tests here cover what we need to test:

In common tests we actually test if metadata is passed when it should, to the underlying objects, and here we don't do that. I think it'd be nice to have specific tests for LRCV in the common tests using the machinary there, even if they're not a part of the _common_ tests as the other ones.

Reply:
The problem with the ones in the common tests was that they were not actually working properly for scorers. The score method does not raise an exception in the inner implementation of MethodMetadataRequest, if we don't set the score request.

Reply:
Can you please write a minimal test for it? It should raise, there shouldn't be a difference between `fit` and `score` in that regard.

Reply:
Could you kindly provide an example of a scorer with LogisticRegressionCV that would raise UnsetMetadataPassedError when sample_weight is passed but the score request is not set?

Reply:
This commit contains the precise part that is related to setting the unrequested params https://github.com/scikit-learn/scikit-learn/pull/24498/commits/f221ee9a0bf7681899851b73e9104eeba5ca5f75

Reply:
generally it's not the _consumer_ which raises the error, rather the _router_ does that when `process_routing` is called. So the only thing we need to test is for the scorer to have the right request values.

Reply:
yes, pretty much, very similar to how we do it in consumer estimators there.

Reply:
@adrinjalali The tests in CircleCI seem to be failing.

Reply:
Seems to have something to do with `set_output` (I might be wrong). I'm checking. cc @thomasjpfan

Reply:
@adrinjalali The checks still seem to be failing but could you kindly check the test that I added in the common tests?